src/main/scala/com/fulcrumgenomics/bam/OverlappingBasesConsensusCaller.scala

```diff
@@ -0,0 +1,250 @@
+        r2BasesBuilder.addAll(r2Bases.slice(from=r2LastReadPos + 1, until=r2Head.read + 1))
+        r2QualsBuilder.addAll(r2Quals.slice(from=r2LastReadPos + 1, until=r2Head.read + 1))
+        r2LastReadPos = r2Head.read
+        r2Iter.next()
+      }
+      else { // matched reference bases, so consensus call
+        // add read bases from insertions
+        r1BasesBuilder.addAll(r1Bases.slice(from = r1LastReadPos + 1, until = r1Head.read))
+        r1QualsBuilder.addAll(r1Quals.slice(from = r1LastReadPos + 1, until = r1Head.read))
+        r2BasesBuilder.addAll(r2Bases.slice(from = r2LastReadPos + 1, until = r2Head.read))
+        r2QualsBuilder.addAll(r2Quals.slice(from = r2LastReadPos + 1, until = r2Head.read))
+
+        r1OverlappingBases += 1
+        r2OverlappingBases += 1
+
+        // consensus call current
+        val (base, qual) = consensusCall(
+          base1 = r1Bases(r1Head.read - 1),
+          qual1 = r1Quals(r1Head.read - 1),
+          base2 = r2Bases(r2Head.read - 1),
+          qual2 = r2Quals(r2Head.read - 1)
+        )
+
+        // Only add the consensus call if we aren't masking disagreements or its a no-call
+        if (onlyMaskDisagreements && base != NoCall) {
+          r1BasesBuilder.addOne(r1Bases(r1LastReadPos))
+          r1QualsBuilder.addOne(r1Quals(r1LastReadPos))
+          r2BasesBuilder.addOne(r2Bases(r2LastReadPos))
+          r2QualsBuilder.addOne(r2Quals(r2LastReadPos))
+        } else {
+          r1BasesBuilder.addOne(base)
+          r1QualsBuilder.addOne(qual)
+          r2BasesBuilder.addOne(base)
+          r2QualsBuilder.addOne(qual)
+        }
+
+        // consume the current
+        r1LastReadPos = r1Head.read
+        r2LastReadPos = r2Head.read
+        r1Iter.next()
+        r2Iter.next()
+      }
+    }
+
+    // add any remaining read and reference bases
+    r1BasesBuilder.addAll(r1Bases.drop(r1LastReadPos))
+    r1QualsBuilder.addAll(r1Quals.drop(r1LastReadPos))
+    r2BasesBuilder.addAll(r2Bases.drop(r2LastReadPos))
+    r2QualsBuilder.addAll(r2Quals.drop(r2LastReadPos))
+
+    require(r1BasesBuilder.length == r1Bases.length)
+    require(r2BasesBuilder.length == r2Bases.length)
+    require(r1QualsBuilder.length == r1Quals.length)
+    require(r2QualsBuilder.length == r2Quals.length)
+
+    r1.bases = r1BasesBuilder.result()
+    r1.quals = r1QualsBuilder.result()
+    r2.bases = r2BasesBuilder.result()
+    r2.quals = r2QualsBuilder.result()
+
+    val r1Corrected = r1Bases.zip(r1.bases).count { case (left, right) => left != right }
+    val r2Corrected = r2Bases.zip(r2.bases).count { case (left, right) => left != right }
+
+    CorrectionStats(r1OverlappingBases, r2OverlappingBases, r1Corrected, r2Corrected)
+  }
+
+  private def consensusCall(base1: Byte, qual1: PhredScore, base2: Byte, qual2: PhredScore): (Byte, PhredScore) = {
+    // Capture the raw consensus base prior to masking it to N, so that we can compute
+    // errors vs. the actually called base.
```

Comment:
I tend to think that if _either_ base is already a no-call then we should probably do _nothing_ as we don't actually have a second base call (not convert the output base to a no-call).  That has ramifications beyond this method though, so perhaps we should only call this if both bases are [ACGT]?